app/assets/javascripts/discourse/app/components/emoji-picker.js

```diff
@@ -243,12 +243,111 @@ export default Component.extend({
     section && section.scrollIntoView();
   },
 
+  numEmojiPerRow(emojis) {
+    // See: https://stackoverflow.com/a/49888033
+    const container = document.querySelector(
+      ".emojis-container .section-group"
+    );
+    const rowLength = Math.floor(container.clientWidth / emojis[0].clientWidth);
+    const totalEmojis = emojis.length;
+    const numElementsLastRow = totalEmojis % rowLength;
+    return rowLength - numElementsLastRow;
+  },
+
   @action
   keydown(event) {
+    const arrowKeys = ["ArrowDown", "ArrowUp", "ArrowLeft", "ArrowRight"];
+    let focusedOnPicker = document.activeElement.closest(
+      ".emoji-picker-emoji-area"
+    )
+      ? document.activeElement
+      : null;
+    let focusedOnEmoji = document.activeElement?.classList.contains("emoji");
```

Comment:
```suggestion
    const focusedOnEmoji = document.activeElement?.classList.contains("emoji");
```

We should use a constant if the value of the variable is never going to change. Might be a few more cases of this here (`focusedOnSearch` etc.).

Reply:
Ah yes, definitely a lot of messy stuff here/there, will be cleaning up the code a lot after improving the traversal logic 😅